MoltenVK/MoltenVK/Commands/MVKMTLResourceBindings.h

```diff
@@ -49,8 +68,27 @@ typedef struct MVKMTLBufferBinding {
     VkDeviceSize offset = 0;
     uint32_t size = 0;
 	uint16_t index = 0;
+    bool justOffset = false;
     bool isDirty = true;
     bool isInline = false;
+
+    inline void markDirty() { justOffset = false; isDirty = true; }
+
+    inline void update(const MVKMTLBufferBinding &other) {
+        if (mtlBuffer != other.mtlBuffer || size != other.size || isInline != other.isInline) {
+            mtlBuffer = other.mtlBuffer;
+            size = other.size;
+            isInline = other.isInline;
+            offset = other.offset;
+
+            justOffset = false;
+            isDirty = true;
+        } else if (offset != other.offset) {
+            offset = other.offset;
+            justOffset = !isDirty || justOffset;
```

Comment:
TBH...I'm having trouble following the logic of when `justOffset` will be set to `true`. Here we seem to be setting it to `true` only if I am not already dirty or `justOffset` was previously set to `true` somehow. 

I may just be missing something in the flow, but it's not clear, in the same way that the values of `isDirty` and `justOffset` are clearly set on each invocation everywhere else.

Reply:
My thinking is that if the binding was updated previously, we can't set it to justOffset here, unless the previous update also only changed the offset. If we did, we wouldn't set the changed buffer from the first update when encoding state.